src/components/Editor/MediaUploader/LocalUploader.jsx

```diff
@@ -25,90 +24,78 @@ const LocalUploader = ({
 }) => {
   const { t } = useTranslation();
 
-  const [pendingUploads, setPendingUploads] = useState([]);
-
-  const uppyConfig = useMemo(
-    () => (isImage ? DEFAULT_IMAGE_UPPY_CONFIG : DEFAULT_VIDEO_UPPY_CONFIG),
-    []
-  );
+  const uploadConfig = isImage
+    ? DEFAULT_IMAGE_UPLOAD_CONFIG
+    : DEFAULT_VIDEO_UPLOAD_CONFIG;
 
   const fileInputRef = useRef(null);
   const dropTargetRef = useRef(null);
 
-  const { uppy, isUploading } = useUppyUploader({ uppyConfig });
+  const { addFiles, uploadFiles, queuedFiles, cancelUpload, isUploading } =
+    useFileUploader({
+      config: uploadConfig,
+      setIsUploadingOnHost: setIsUploading,
+    });
 
   setIsUploading(isUploading);
 
-  const handleAddFile = ({ target: { files } }) => {
-    Array.from(files).forEach(file => {
-      try {
-        uppy.addFile({
-          name: file.name,
-          type: file.type,
-          data: file,
-        });
-      } catch (error) {
-        if (error.message !== t("neetoEditor.error.onBeforeFileAddedReturn")) {
-          Toastr.error(t("neetoEditor.error.cannotAddFiles"));
-        }
-      }
-    });
-
-    afterAddingFiles();
+  const handleAddFile = async ({ target: { files } }) => {
+    addFiles(files);
+    const uploadedFiles = await uploadFiles();
+    uploadedFiles.forEach(insertMediaToEditor);
+    onClose();
   };
 
-  const afterAddingFiles = () => {
-    const newlyAddedFiles = uppy.getFiles().map(file => ({
-      id: file.id,
-      filename: file.name,
-      signedId: "awaiting",
-      url: "",
-      progress: 0,
-    }));
-    if (isEmpty(newlyAddedFiles)) {
-      uppy.reset();
-
-      return;
-    }
-    setPendingUploads(concat(newlyAddedFiles));
-    handleUpload();
-  };
+  useEffect(() => {
+    const dropZone = dropTargetRef?.current;
+    let isDragging = false;
```

Comment:
For both `LocalUploader.jsx` and `Attachments.jsx` files, can you move all drag and drop related logic to a `hooks/useDragAndDrop.jsx` in the same folder? It need not be reused. It will clean up the code.